File: ClickDigitalBackend/src/main/java/rulemanager/models/Trigger.java
package rulemanager.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

/**

 * An object of this class represents a internal trigger.
 */
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Trigger {

    public String name = "";
    public int ID = 0;
    public String triggerclass = "";
    public String deviceID = "";
    public String sensorID = "";
    public Condition condition = null;

    public Trigger(){}
}

File: ClickDigitalFrontend/src/app/models/frontend/rule_module/trigger.ts
import { Condition } from './condition';

// Trigger template for matching with JSON Object  from frontend received by backend
export class Trigger {

  ID = 0;
  name = ''; // to save trigger this value has not to be null or ''
  triggerclass = '';
  deviceID = ''; // to save trigger this value has not to be null or ''
  sensorID = ''; // to save trigger this value has not to be null or ''
  condition: Condition = new Condition();
}
